frontend/packages/process-editor/src/classes/ProcessTaskManager/RemoveProcessTaskManager.ts

```diff
@@ -0,0 +1,77 @@
+import { type BpmnApiContextProps } from '../../contexts/BpmnApiContext';
+import { type BpmnDetails } from '../../types/BpmnDetails';
+import { type TaskEvent } from './types';
+import { BpmnTypeEnum } from '../../enum/BpmnTypeEnum';
+import {
+  getDataTypeIdFromBusinessObject,
+  getLayoutSetIdFromTaskId,
+} from '../../utils/hookUtils/hookUtils';
+import { type Policy } from '../../utils/policy/types';
+import { PaymentPolicyBuilder } from '../../utils/policy';
+
+export class RemoveProcessTaskManager {
+  constructor(
+    private readonly org: string,
+    private readonly app: string,
+    private readonly layoutSets: BpmnApiContextProps['layoutSets'],
+    private readonly deleteLayoutSet: BpmnApiContextProps['deleteLayoutSet'],
+    private readonly deleteDataTypeFromAppMetadata: BpmnApiContextProps['deleteDataTypeFromAppMetadata'],
+    private readonly bpmnDetails: BpmnDetails,
+    private readonly currentPolicy: Policy,
+  ) {}
+
+  public handleTaskRemove(taskEvent: TaskEvent): void {
+    if (this.bpmnDetails.type === BpmnTypeEnum.Task) {
+      this.handleDataTaskRemove();
+    }
+
+    if (this.bpmnDetails.taskType === 'payment') {
+      this.handlePaymentTaskRemove(taskEvent);
+    }
+
+    if (this.bpmnDetails.taskType === 'signing') {
+      this.handleSigningTaskRemove(taskEvent);
+    }
+  }
+
+  private handleDataTaskRemove(): void {
```

Comment:
Adding the comment here, but it applies to all these private methods, and in the AddProcessTaskManager as well:
Should we document which actions we expect to trigger for each of the methods? F.ex. for data task, we expect to add/remove associated layoutset. For payment taks, we expect to add/remove associated layoutset, datatype and policy rule.
Documenting this on the method ensures that we are clearer on what we expect, and review would also be easier in the future as one could review that the code performs all the expected actions based on the description 😊

Reply:
Yes, great suggestion and I do agree! 🥳

Reply:
Done ✅